### PR TITLE
[MAIN] [STRATCONN] added consent for google campaign manager

### PR DESCRIPTION
**Pull request by AnkitSegment**
Added consent mode settings for google campaign manager

JIRA TICKET: https://segment.atlassian.net/browse/STRATCONN-3789

Action Tester:

Screenshot 2024-05-02 at 3 27 47 PM

Segment App:

Screenshot 2024-05-10 at 2 06 02 PM

GTAG manager:

Screenshot 2024-05-10 at 2 03 05 PM

Chrome console:

Screenshot 2024-05-10 at 2 04 56 PM

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [X] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [X] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by AnkitSegment** [comment]
Hey @danieljackins, Changed size-limit from 151 KB to 152 KB because builds are failing. 
 
Screenshot 2024-05-10 at 2 30 53 PM

**Review by harsh-joshi99 — APPROVED** [review]
LGTM ✨

**Review by Innovative-GauravKochar — DISMISSED** [review]
Nice Work @AnkitSegment  ,Looks good to me

File: packages/browser-destinations/destinations/google-campaign-manager/src/__tests__/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    partnerAction: 'salesActivity',
+    name: 'Sales Activity',
+    enabled: true,
+    subscribe: 'type = "track"',
+    mapping: {}
+  }
+]
+
+describe('Google Tag for Campaign Manager', () => {
+  const defaultSettings = {
```
Comment (varadarajan-tw) [inline packages/browser-destinations/destinations/google-campaign-manager/src/__tests__/index.test.ts:16]: could you also pls add a test for `enableConsentMode` not being set?

**Review by varadarajan-tw — COMMENTED** [review]
LGTM! Just add a suggestion for test case.

File: packages/browser-destinations/destinations/google-campaign-manager/src/__tests__/index.test.ts (hunk 1 above)
Comment (AnkitSegment) [reply packages/browser-destinations/destinations/google-campaign-manager/src/__tests__/index.test.ts:16]: added.

**Review by varadarajan-tw — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @AnkitSegment PR deployed

**Pull request merged**